patches/lua-tree.patch

```diff
@@ -373,26 +465,26 @@ index c1a150a..3d6c639 100644
 -
 -
  local typedefs = {}
-
-
-@@ -385,13 +364,13 @@ typedefs.sni = Schema.define {
-
+ 
+ 
+@@ -386,13 +365,13 @@ typedefs.sni = Schema.define {
+ 
  typedefs.certificate = Schema.define {
    type = "string",
 -  custom_validator = validate_certificate,
 +  custom_validator = x509.validate_certificate,
  }
-
-
+ 
+ 
  typedefs.key = Schema.define {
    type = "string",
 -  custom_validator = validate_key,
 +  custom_validator = x509.validate_key,
  }
-
-
+ 
+ 
```

Comment:
there are several of these lines with whitespace.  i think cleaning them would make the diff much easier to read

Reply:
These whitespaces were generated by the Git patch generation itself, I did not purposefully add these.

Whitespaces in patch files shouldn't be edited as these are part of the source files & how the diff was generated, if this were to be edited, the patch wouldn't be able to be applied without telling it to ignore whitespace (via `git apply --reject --whitespace=fix`).

Reply:
right, no need to edit whitespaces (even if only because git is particularly bad in that respect), but this commit *adds* a lot of them. they weren't there before.

Reply:
Correct, I assume because the past author of these changes had their editor strip the whitespaces. I didn't edit these manually, I just pasted in the generated patch & rearranged two files as they were previously out of order (and I left it that way).

The way the patch is after these changes is how it actually should have been in the first place.